localstack-core/localstack/services/kms/provider.py

```diff
@@ -1354,6 +1357,74 @@ def enable_key_rotation(
             key.rotation_period_in_days = request.get("RotationPeriodInDays")
         key._update_key_rotation_date()
 
+    @handler("ListKeyRotations", expand=False)
+    def list_key_rotations(
+        self, context: RequestContext, request: ListKeyRotationsRequest
+    ) -> ListKeyRotationsResponse:
+        # https://docs.aws.amazon.com/kms/latest/developerguide/key-state.html
+        # "If the KMS key has imported key material or is in a custom key store: UnsupportedOperationException."
+        # We do not model that here, though.
+        # TODO: Add support for single-region EXTERNAL key origins
+        key = self._get_kms_key(
+            context.account_id,
+            context.region,
+            request.get("KeyId"),
+            disabled_key_allowed=True,
+            pending_deletion_key_allowed=True,
+        )
+
+        include_key_material = request.get("IncludeKeyMaterial")
+        if include_key_material is not None:
+            if include_key_material not in ["ALL_KEY_MATERIAL", "ROTATIONS_ONLY"]:
+                raise ValidationException(
+                    f"1 validation error detected: Value '{include_key_material}' at 'includeKeyMaterial' "
+                    f"failed to satisfy constraint: Member must satisfy enum value set: [ALL_KEY_MATERIAL, ROTATIONS_ONLY]"
+                )
+
+            if key.metadata["KeySpec"] != KeySpec.SYMMETRIC_DEFAULT:
+                raise ValidationException(
+                    "Input parameter includeKeyMaterial can only be used with symmetric keys."
+                )
+
+            self._validate_key_supports_include_key_material(key)
+
+        rotation_history = []
+        if key.supports_rotation() or include_key_material is not None:
```

Comment:
Why do we validate similar checks twice here: `_validate_key_supports_include_key_material()` and `supports_rotation()`?